### ISSUE TITLE
Engines related suggestion in Batch downloader

### ISSUE THREAD

**Issue by zinalili** [mhogomchungu/media-downloader#390]
1. Please make it so that the engine with which the URL was added is respected.
For example, if I paste an URL in "Enter URL" field, select Aria2 and then click on "Add URL in list", I expect this URL to be downloaded with Aria2, no matter what engine is actually set at the time of clicking "Download".
2. You have already created a context option for each download which is called "Engine name". Which allows to choose an engine for the specific entry. Plus, when used, each entry shows the engine with which it will be downloaded. Please, make it default behavior. That will be useful if you implement my first suggestion.
Thank you!

**Comment by mhogomchungu** [comment]
Done but its not default behavior because it unnecessarily clutter the UI for those who do not want this behavior,

When the new version is out, enable the behavior you want adding "AutoSetDefaultEngineAndOptions=true" option is the settings file.

The settings file is located in a setting folder that is located in the "app data path" mentioned in the basic downloader tab when the app starts.

**Comment by zinalili** [comment]
Thank you but it does not seem like it works as expected. I changed "false" to "true" in abovementioned string in the settings file. Updated to 4.7.0 but it works just like before.
Here is what I do:
1. Go to batchdownloader tab.
2. Paste the URL: https://www.youtube.com/watch?v=xuP4g7IDgDM (it's a 10 sec video)
3. Download options: -f bestaudio -x --embed-thumbnail
4. Click on "Add URL to list"
5. Right click on the video, then "Show media options", then go to the very bottom, pick "22" and click "Set"
6. Click on "Download"
The result is that the video gets downloaded as "22" but then post-processed into "-f bestaudio -x --embed-thumbnail"

**Comment by mhogomchungu** [comment]
>Please make it so that the engine with which the URL was added is respected.
For example, if I paste an URL in "Enter URL" field, select Aria2 and then click on "Add URL in list", I expect this URL to be downloaded with Aria2, no matter what engine is actually set at the time of clicking "Download".

You asked for the above where you asked for default engine to be set when an entry is added to the UI, can you confirm the above work?

Now, you are asking about what downloading option to be effective when conflicting options are set from two different places. Go to config file and then change
```DownloadOptionsAsLast``` option to true.

**Comment by zinalili** [comment]
> You asked for the above where you asked for default engine to be set when an entry is added to the UI, can you confirm the above work?

Yes, it does.

> `DownloadOptionsAsLast` option to true

Done but no effect so far. I used the same link from my previous post. Added it with "-f bestaudio -x --embed-thumbnail", changed it to "22" from "Show media options", then downloaded it and it still was downloaded as "-f bestaudio -x --embed-thumbnail".
As an example, why this is useful: when I was testing it just now, the youtube video wouldn't download at first try throwing error "format is not available" (sometimes this happens with youtube). I added a video with "22" format, got an error, right-clicked on it and selected "Download". The video is then downloaded but with the current "global" parameters, not with the one the video was added.
